router.spec.js

```diff
@@ -1,18 +1,141 @@
 const chai = require('chai');
 const chaiHttp = require('chai-http');
 
+process.env.NODE_ENV = 'test';
+
 const server = require('./index');
+const knex = require('./db/db');
+const user = require('./db/models/user');
+const repository = require('./db/models/repository');
+const contribution = require('./db/models/contribution');
 
 const should = chai.should();
 
 chai.use(chaiHttp);
 
+const mockRepo = {
+  id: 1,
+  owner: 1,
+  full_name: 'Bela Todo list',
+  description: 'Bela Todo list',
+  html_url: 'https://github.com/',
+  language: 'eng',
+  stargazers_count: 5
+};
+
+const mockUser = {
+  id: 1,
+  login: 'bela',
+  avatar_url: 'https://github.com/avatar',
+  html_url: 'https://github.com/avatar'
+};
+
+const mockContribution = {
+  user: 1,
+  repository: 1,
+  line_count: 50
+};
+
 describe('Web instance', () => {
-    describe('GET /hello', () => {
-      it('returns \'Hello World ! \'', async () => {
-        const response = await chai.request(server).get('/hello');
+  beforeEach(async () => {
+    await knex.raw('TRUNCATE TABLE contribution CASCADE');
+    await knex.raw('TRUNCATE TABLE repository CASCADE');
+    await knex.raw('TRUNCATE TABLE "user" CASCADE');
+    await knex.raw('ALTER SEQUENCE repository_id_seq RESTART WITH 1');
```

Comment:
Wouldn't it be easier to have a migration down and up script before running the tests?

Reply:
I was unable to execute the migration script from the router.spec.js file. If I run it from the console it works fine.

Reply:
Yes, it's preferable to have a separate script that can be run on the console. And since `test-web` is an npm script as well they can be combined with the `&&` bash syntax. For example the test script could look like: `npm run reset-db && npm run test` where `reset-db` calls the down and up migration functions, and possible a seed as well.